client/service/src/lib.rs

```diff
@@ -92,13 +93,13 @@ pub struct Service<TBl, TCl, TSc, TNetStatus, TNet, TTxPool, TOc> {
 	/// A receiver for spawned essential-tasks concluding.
 	essential_failed_rx: mpsc::UnboundedReceiver<()>,
 	/// Sender for futures that must be spawned as background tasks.
-	to_spawn_tx: mpsc::UnboundedSender<Box<dyn Future<Item = (), Error = ()> + Send>>,
+	to_spawn_tx: mpsc::UnboundedSender<Box<dyn Future<Output = ()> + Send + Unpin>>,
 	/// Receiver for futures that must be spawned as background tasks.
-	to_spawn_rx: mpsc::UnboundedReceiver<Box<dyn Future<Item = (), Error = ()> + Send>>,
+	to_spawn_rx: mpsc::UnboundedReceiver<Box<dyn Future<Output = ()> + Send + Unpin>>,
```

Comment:
I'm not going to comment on every single one, but you get the idea. `Box<... + Unpin>` should be replaced with `Pin<Box<...>>`.